lib/zig.h

```diff
@@ -185,7 +185,7 @@ typedef char bool;
 #elif defined(__i386__) || defined(__x86_64__)
 #define zig_trap() __asm__ volatile("ud2");
 #else
-#define zig_trap() raise(SIGILL)
+#define zig_trap() raise(SIGTRAP)
```

Comment:
I've previously explained my reasoning for SIGILL but now I think SIGILL is not much better or as good as SIGTRAP, anyway.

After some more research, now I would like to give out my recommendation to maybe use SIGSTOP or SIGKILL here as they are AFAIK the only signals that **cannot be caught** which I believe would be more appropriate here? I was going to demonstrate how they cannot be caught but these two asserts are perhaps demonstrative enough:
https://github.com/ziglang/zig/blob/e7f128c2051b086cdb1c03da041745b560bbaa3e/lib/std/os/linux.zig#L1160-L1161

Raising signals that cannot be caught might be more appropriate as `trap`/`@trap` is not a debugging tool but rather something that should in fact terminate the program for sure, as far as I understand it.

The difference between SIGSTOP and SIGKILL seems to be that SIGSTOP actually stops the program for later resumption which seems inappropriate so perhaps SIGKILL is best of all here.